python/cudf_polars/tests/test_translate_in_debug_mode.py

```diff
@@ -0,0 +1,21 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import pytest
+
+import polars as pl
+
+
+def test_cudf_polars_debug_mode():
+    df = pl.LazyFrame(
+        {"key": [1, 1, 1, 2, 3, 3, 2, 2], "value": [1, 2, 3, 4, 5, 6, 7, 8]}
+    )
+    q = df.select(pl.col("value").sum().over("key"))
+
+    with pytest.raises(
+        Exception,
+        match=r"(NotImplementedError|Query contained unsupported operations|Grouped rolling window not implemented)",
+    ):
+        q.collect(engine=pl.GPUEngine(debug_mode=True, raise_on_fail=True))
+    # assert_ir_translation_raises(q, NotImplementedError)
```

Comment:
I'm fine with using `assert_ir_translation_raises` instead but I think I'd just end up stuffing the regex match in `assert_ir_translation_raises`. I'm doing this because regardless of the `Exception` we raise in `callback.py`, Polars will raise its own `ComputeError` (I think its because it actually executes the IR graph). Is there anyway to catch the original error and handle it?